### PR TITLE
Add parent ID query

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Add parent ID query

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by vagimeli — APPROVED** [review]
LGTM

File: _query-dsl/joining/index.md [hunk 1]
```diff
@@ -12,10 +12,13 @@ redirect_from:
...
 
-- `nested` queries: Act as wrappers for other queries to search [nested]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/nested/) fields. The nested field objects are searched as though they were indexed as separate documents.
-- [`has_child`]({{site.url}}{{site.baseurl}}/query-dsl/joining/has-child/) queries: Search for parent documents whose child documents match the query.
-- [`has_parent`]({{site.url}}{{site.baseurl}}/query-dsl/joining/has-parent/) queries: Search for child documents whose parent documents match the query.
-- `parent_id` queries: A [join]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/join/) field type establishes a parent/child relationship between documents in the same index. `parent_id` queries search for child documents that are joined to a specific parent document. 
+
+- Queries to search [nested]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/nested/) fields:
```
Comment (natebower) [inline _query-dsl/joining/index.md:16]: "to search" should either be "for searching" or "that search".

File: _query-dsl/joining/index.md [hunk 2]
```diff
@@ -12,10 +12,13 @@ redirect_from:
...
+    - `nested` queries: Act as wrappers for other queries to search [nested]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/nested/) fields. The nested field objects are searched as though they were indexed as separate documents.
+- Queries to search documents connected by a [join]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/join/) field type, which establishes a parent/child relationship between documents in the same index:
```
Comment (natebower) [inline _query-dsl/joining/index.md:18]: Same

File: _query-dsl/joining/parent-id.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
+---
+layout: default
+title: Parent ID
+parent: Joining queries
+nav_order: 40
+---
+
+# Parent ID query
+
+The `parent_id` query returns child documents whose parent document has the specified ID. You can establish parent-child relationships between documents in the same index by using a [join]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/join/) field type.
```
Comment (natebower) [inline _query-dsl/joining/parent-id.md:10]:
```suggestion
The `parent_id` query returns child documents whose parent documents have the specified ID. You can establish parent-child relationships between documents in the same index by using a [join]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/join/) field type.
```

File: _query-dsl/joining/parent-id.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
```
Comment (natebower) [inline _query-dsl/joining/parent-id.md:11]: Above, global: We use "parent/child" (forward slash) above but "parent-child" here. Should be consistent globally.

File: _query-dsl/joining/parent-id.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %} 
+
+For this example, first configure an index that contains documents representing products and their brands as described in the [`has_child` query example]({{site.url}}{{site.baseurl}}/query-dsl/joining/has-child/). 
+
+To search for children documents of a specific parent document, use a `parent_id` query. The following query returns child documents (products) whose parent document has the ID `1`:
```
Comment (natebower) [inline _query-dsl/joining/parent-id.md:35]:
```suggestion
To search for child documents of a specific parent document, use a `parent_id` query. The following query returns child documents (products) whose parent document has the ID `1`:
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

File: _query-dsl/joining/index.md (hunk 1 above)
Comment (kolchfa-aws) [inline _query-dsl/joining/index.md:16]:
```suggestion
- Queries for searching [nested]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/nested/) fields:
```

File: _query-dsl/joining/index.md (hunk 2 above)
Comment (kolchfa-aws) [inline _query-dsl/joining/index.md:18]:
```suggestion
- Queries for searching documents connected by a [join]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/join/) field type, which establishes a parent/child relationship between documents in the same index:
```

File: _query-dsl/joining/parent-id.md (hunk 3 above)
Comment (kolchfa-aws) [reply _query-dsl/joining/parent-id.md:10]: The parent:child is a 1:many relationship so leaving singular.

**Pull request merged**